### PR TITLE
use ember-macro-helpers for computed

### PR DESCRIPTION
**Pull request by kellyselden**
This is a refactor to use the same base library as ember-moment, ember-computed-decorators, and ember-awesome-macros for composable macros.

Caveats:
* It eagerly calculates properties (like the ember-computed-decorators model) rather than lazily like you had it. For example, you were able to call `getValue` on the first key, then early return if it returns undefined without calling `getValue` on the second key. This implementation calculates all keys prior to giving them to you. (I have ideas on how to improve this in the future, perhaps an alternate API that passes you the `getValue` function to calculate yourself.)
* The changes needed in the tests outline the minor breaking changes that would come out of this.
  * `Ember.computed.alias` is a unique macro not derived from `Ember.ComputedProperty` making it harder to detect. I decided to not try to check for it in my implementation, because any real-world code doesn't need an alias as a macro parameter.
  * The `object.get('myComputedPromise').foo` to `object.get('myComputedPromise.foo')` change I didn't dig into, but it was needed to fix the tests.

Benefits:
* Shared library
* Big reduction on your code footprint
* All your macros are now composable.
* All your macros now support brace expansion, meaning you could do `sum('foo.bar.{x,y,z}')` or `ifNull('foo.bar.{useThis,orThis}')`.
* The library supports array accessors `.@each` and `.[]`, but I don't think any of your current macros would benefit from this.

Accepting this refactor would mean a major version update at the most, or a minor version at the least. It's unfortunate it's not a perfect refactor. I'm just trying to get everybody on the same macro infrastructure.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +52,8 @@
     "cpm"
   ],
   "dependencies": {
-    "ember-cli-babel": "^5.1.7"
+    "ember-cli-babel": "^5.1.7",
+    "ember-macro-helpers": "kellyselden/ember-macro-helpers#ember-cpm"
```
Comment (kellyselden) [inline package.json]: I had to make an API addition to support this refactor. If you approve, I will release a new version before merge.

**Comment by cibernox** [comment]
Code-wise this is great. Can you rebase now that I've merged the PR that fixes the tests?

**Comment by cibernox** [comment]
Thanks! I have to update the docs first but then I can release a mayor version

**Comment by kellyselden** [comment]
Thank you. I still have to get you off my fork. I will do that today.

**Comment by cibernox** [comment]
I'll wait for that before the release


2017-01-20 16:40 GMT+00:00 Kelly Selden <notifications@github.com>:

**Pull request merged**